datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1470,19 +1470,25 @@ impl<'a, S: SimplifyInfo> TreeNodeRewriter for Simplifier<'a, S> {
             }) => Transformed::yes(simplify_regex_expr(left, op, right)?),
 
             // Rules for Like
-            Expr::Like(Like {
-                expr,
-                pattern,
-                negated,
-                escape_char: _,
-                case_insensitive: _,
-            }) if !is_null(&expr)
-                && matches!(
-                    pattern.as_ref(),
-                    Expr::Literal(ScalarValue::Utf8(Some(pattern_str))) if pattern_str == "%"
-                ) =>
-            {
-                Transformed::yes(lit(!negated))
+            Expr::Like(ref like_expr) => {
+                if let Expr::Literal(ScalarValue::Utf8(Some(pattern_str))) =
+                    like_expr.pattern.as_ref()
+                {
+                    if !is_null(&like_expr.expr) && pattern_str == "%" {
+                        Transformed::yes(lit(!like_expr.negated))
```

Comment:
This code was already here (see diff above). Did I change the logic in any way?

Reply:
Ack. Can you please confirm or disconfirm the correctness concern?

Reply:
I agree this is incorrect but this is the logic that was there before this PR. I'm happy to remove it. I guess no one has complained because the case that breaks is `SELECT NULL LIKE '%'` which isn't something that happens under normal usage.

Reply:
> I agree this is incorrect but this is the logic that was there before this PR.

thanks for confirming and understood it's pre-existing.

Can it be fixed though?
i personally would prefer that `git annotate` on the file wouldn't point at me as the author of something that's known to be incorrect.

Reply:
My latest commit removes that special case. I'm not sure how to fix it, your comment suggests it can't be fixed right?

Reply:
> I guess no one has complained because the case that breaks is `SELECT NULL LIKE '%'` which isn't something that happens under normal usage.

actually the opposite.
this was correct only for this case
and wrong for all other `... LIKE '%'` queries.

see https://github.com/apache/datafusion/pull/13259